### ISSUE TITLE
Disco Enemy

### ISSUE THREAD

**Issue by CTxCB** [josefnpat/VividCast#13]
In the Demo, there's an rotating enemy that flashes blue and green. Unsure if this is a bug or a feature, but it happens every time I try and load the demo... The enemy is to the right of the doorway on the side opposite to where the player(s) spawn.

**Comment by josefnpat** [comment]
Confirmed, thanks
![vividcast_13](https://user-images.githubusercontent.com/646162/32385110-1bf31fe2-c08b-11e7-928b-5c9cd53d5b76.gif)

**Comment by pablomayobre** [comment]
This happened before my change, I have no clue what it is

**Comment by josefnpat** [comment]
Looks like two entities are fighting for zindex order. I think it's a demo example problem.

**Comment by pablomayobre** [comment]
Yeah, won't happen in normal code, unless you have two things occupying the same place

**Comment by CTxCB** [comment]
I've attempted to fix this bug in my Pull Request #15. How I've done this is... Created a table called enemyPossiblePositions with 11 tables within (matching the value of map_width) containing xPos and yPos Data which is fed into the enemy positions by picking one of the 11 tables at random, then removing that table from enemyPossiblePositions once the xPos and yPos values have been used up. This does mean that the enemies have fixed positions, but they're still randomly assigned to a position.

**Comment by pablomayobre** [comment]
You could probably had fixed it by using `love.math.random` instead of `math.random`. Since LÖVE doesn't feed the random seed, the first few values of the random generator are pretty similar (if not equal)

Well this proposed fix would be kinda hacky and your solution is probably final (and better)

**Comment by CTxCB** [comment]
@Positive07 I did the suggestions you mentioned about switching from `math.random` to `love.math.random`, because I still had a few things to clean up here or there. I think I'm at the stage where my Pull Request is ready for @josefnpat to pull it into the master.
